### PR TITLE
Csweaver/annotations

### PR DESCRIPTION
**Pull request by csweaver**
https://docs.google.com/document/d/1Fxjp1SKtCk7l8QP9-7KAjGXL0eldi_qEnNT0NmlGzXI/edit#heading=h.257kxoxyyt0a

Add GET route and scanpy code to fetch annotations for all observations. Can limit the annotations fetched to only certain columns using query params.

### CODE REVIEW

File: server/app/rest_api/rest.py [hunk 1]
```diff
@@ -128,10 +128,53 @@ def get(self):
...
 
+class AnnotationsObsAPI(Resource):
+    @swagger.doc({
+        "summary": "Fetch annotations (metadata) for all observations.",
+        "tags": ["annotations"],
+        "parameters": [{
+            "in": "query",
+            "name": "annotation-keys",
+            "type": "string",
+            "description": "comma-separated annotation keys, ex: num_genes,percent_mito"
```
Comment (ttung) [inline server/app/rest_api/rest.py]: why not make this a list of strings?  that would scale better if people start doing keys that need escaping.

File: server/app/scanpy_engine/scanpy_engine.py [hunk 2]
```diff
@@ -174,38 +174,21 @@ def _filter_annotation(self, filter, index, axis):
...
-                    "range": {
-                        "min": df.obs[field].min(),
-                        "max": df.obs[field].max()
-                    }
-                }
-        return metadata_ranges
-
-    @cache.memoize()
-    def metadata(self, df, fields=None):
+    def annotation(self, df, fields=None):
```
Comment (ttung) [inline server/app/scanpy_engine/scanpy_engine.py]: does anyone use the fields param?

File: server/app/scanpy_engine/scanpy_engine.py [hunk 3]
```diff
@@ -174,38 +174,21 @@ def _filter_annotation(self, filter, index, axis):
...
         """
-         Gets metadata key:value for each cells
+         Gets annotation value for each observation
 
         :param df: from filter_cells, dataframe
-        :param fields: list of keys for metadata to return, returns all metadata values if not set.
-        :return: list of metadata values
+        :param fields: list of keys for annotation to return, returns all annotation values if not set.
+        :return: dict: names - list of fields in order, data - list of lists or metadata [idx, val1, val2...]
```
Comment (ttung) [inline server/app/scanpy_engine/scanpy_engine.py]: idx = index of the observation?

File: server/app/scanpy_engine/scanpy_engine.py (hunk 2 above)
Comment (csweaver) [reply server/app/scanpy_engine/scanpy_engine.py]: yep, that's what is coming from the annotation-names filter

File: server/app/scanpy_engine/scanpy_engine.py (hunk 3 above)
Comment (csweaver) [reply server/app/scanpy_engine/scanpy_engine.py]: yes, will clarify

File: server/app/rest_api/rest.py (hunk 1 above)
Comment (csweaver) [reply server/app/rest_api/rest.py]: good point, I made the suggestion in the requirements doc, waiting on approval

**Pull request merged**